alibi_detect/utils/keops/kernels.py

```diff
@@ -115,13 +117,36 @@ def forward(self, x: LazyTensor, y: LazyTensor, infer_sigma: bool = False) -> La
             kernel_mat = kernel_mat.sum(-1) / len(self.sigma)
         return kernel_mat
 
+    def get_config(self) -> dict:
+        """
+        Returns a serializable config dict (excluding the input_sigma_fn, which is serialized in alibi_detect.saving).
+        """
+        cfg = self.config.copy()
+        if isinstance(cfg['sigma'], torch.Tensor):
+            cfg['sigma'] = cfg['sigma'].detach().cpu().numpy().tolist()
```

Comment:
Are we certain we don't need a deep copy here (depends on the dict values)? Wouldn't be ideal if `self.config` turns out to not be idempotent (different depending whether `get_config` has been called or not.

Reply:
I guess the same question applies to every component for which we have `self.config` and which may have mutable values - do we need to do an audit / have tests?

Reply:
Upon review of this I think you're right. We seem to get away with it in our tests (and in the informal tests I did previously) because when we update items in `config` we are typically redefining the entire item, rather than mutating it (e.g. we change `sigma` from `None` to a `Tensor`, rather than mutating an element in `sigma`). I think this means the shallow copy is OK since the reference is broken, e.g.:

```python
orig_a = [1,2,3]
dict1 = {'a': deepcopy(orig_a), 'b': 'cat'}
dict2 = dict1.copy()

dict1['a'] = [1,1,1]

dict2['a'] == orig_a

>>> True
```

However, as you say, if we were to mutate an item in `config`, we would have a problem:
```python
orig_a = [1,2,3]
dict1 = {'a': deepcopy(orig_a), 'b': 'cat'}
dict2 = dict1.copy()

dict1['a'][1] = 1 

dict2['a'] == orig_a

>>> False
```

I might be missing something but seems we should add `deepcopy` to all of these. I can update these ones in this PR and then do a follow-up for the rest of them?

Reply:
Yea I think better safe than sorry.

Reply:
Tracker: https://github.com/SeldonIO/alibi-detect/issues/683

Reply:
Done for keops kernel only in 669a8c60bc4cdb1bdd6e6f6ead9008bfa7d2f3a2